enumeratum-core/src/main/scala/enumeratum/EnumEntry.scala

```diff
@@ -61,4 +61,17 @@ object EnumEntry {
     abstract override def entryName: String = super.entryName.toLowerCase
   }
 
+  /**
+   * Helper implicit for more readable checking.
+   */
+  implicit class ComparableEnum[A <: EnumEntry](enum: A) {
```

Comment:
The purpose of this type constraint is to restrict the implicit class to apply only to EnumEntry subtypes. I think it has only role in compile time, but it seems to be necessary.